lxd/api_cluster.go

```diff
@@ -117,39 +174,43 @@ func clusterPutJoin(d *Daemon, req api.ClusterPut) Response {
 	if len(req.ClusterCertificate) == 0 {
 		return BadRequest(fmt.Errorf("No target cluster node certificate provided"))
 	}
+
 	address, err := node.HTTPSAddress(d.db)
 	if err != nil {
 		return SmartError(err)
 	}
+
 	if address == "" {
-		return BadRequest(fmt.Errorf("No core.https_address config key is set on this node"))
-	}
+		if req.ServerAddress == "" {
+			return BadRequest(fmt.Errorf("No core.https_address config key is set on this node"))
+		}
 
-	// Get all defined storage pools and networks, so they can be compared
-	// to the ones in the cluster.
-	pools := []api.StoragePool{}
-	poolNames, err := d.cluster.StoragePools()
-	if err != nil && err != db.ErrNoSuchObject {
-		return SmartError(err)
-	}
-	for _, name := range poolNames {
-		_, pool, err := d.cluster.StoragePoolGet(name)
+		// The user has provided a server address, and no networking
+		// was setup on this node, let's do the job and open the port.
+		err := d.db.Transaction(func(tx *db.NodeTx) error {
+			config, err := node.ConfigLoad(tx)
+			if err != nil {
+				return errors.Wrap(err, "failed to load cluster config")
```

Comment:
Errors must start with a capital (and never end with a period)

Reply:
Fixed.